spec/rubocop/cop/style/mutable_constant_spec.rb

```diff
@@ -43,6 +43,43 @@
     end
   end
 
+  shared_examples 'literals that are frozen' do |o|
+    let(:prefix) { o }
+
+    it_behaves_like 'immutable objects', '[1, 2, 3]'
+    it_behaves_like 'immutable objects', '%w(a b c)'
+    it_behaves_like 'immutable objects', '{ a: 1, b: 2 }'
+    it_behaves_like 'immutable objects', "'str'"
+    it_behaves_like 'immutable objects', '"top#{1 + 2}"'
+    it_behaves_like 'immutable objects', '1'
+    it_behaves_like 'immutable objects', '1.5'
+    it_behaves_like 'immutable objects', ':sym'
+    it_behaves_like 'immutable objects', 'FOO + BAR'
+    it_behaves_like 'immutable objects', 'FOO - BAR'
+    it_behaves_like 'immutable objects', "'foo' + 'bar'"
+    it_behaves_like 'immutable objects', "ENV['foo']"
+    it_behaves_like 'immutable objects', "::ENV['foo']"
```

Comment:
I don't understand these. `X = FOO + BAR` should raise an offense (unless `experimental_*`), as they won't be frozen in general, e.g.:

```ruby
FOO = [1, 2, 3].freeze
BAR = [1, 2, 3].freeze
X = FOO + BAR
X.frozen? # => false
```

Reply:
Yes @marcandre - I guess you are right,  tt should raise an offense. Looks like, in the current behaviour it does not. Can you check the following tests:
https://github.com/rubocop/rubocop/blob/master/spec/rubocop/cop/style/mutable_constant_spec.rb#L58

Reply:
Found the issue - 
The mutable constant cop relies on [unless value.mutable_literal?](https://github.com/rubocop/rubocop/blob/master/lib/rubocop/cop/style/mutable_constant.rb#L122) whose definition is [here](https://github.com/rubocop/rubocop-ast/blob/master/lib/rubocop/ast/node.rb#L361). This is actually not used properly. 

@marcandre  - I think we should create another bug for this and solve it for this cop.

Reply:
Oh, right. It's ok to not change the current behaviour then, and yes, it could be discussed elsewhere 👍